### DISCUSSION TITLE
How do I get the Virtual ZPL Printer to show up in my Printers & scanners?

### DISCUSSION THREAD

**Discussion by bhorderstuff** [porrey/Virtual-ZPL-Printer#39] [Q&A]
How do I get the Virtual ZPL Printer to show up in my Printers & scanners in Windows 10?

**Comment by porrey** [comment]
This virtual printer is designed for applications that make a direct TCP/IP connection to a printer that understands ZPL command. It was not intended to work as a Windows based printer.

**Reply by bhorderstuff** [reply]
Thanks for the input.
Unfortunately, I'm working with a program that looks for the Windows printer and then sends the ZPL code.
I can't seem to find a workaround.

**Comment by GozerBCN** [comment]
> This virtual printer is designed for applications that make a direct TCP/IP connection to a printer that understands ZPL command. It was not intended to work as a Windows based printer.

Hello,
What about using a downloaded driver from:
https://www.*****.com/us/en/support-downloads/printers.html (zebra)

adding a TCP/IP print port in Windows (pointing to your sw ip:port ) then using one of the Zebra Print drivers for Win to print?

**Reply by Mabenan** [reply]
Thats exactly hiw it works i recoomebd the zebra print utility software for adding and the driver of the zt420 203dpi

**Reply by rafamuz** [reply]
> > This virtual printer is designed for applications that make a direct TCP/IP connection to a printer that understands ZPL command. It was not intended to work as a Windows based printer.
> 
> Hello, What about using a downloaded driver from: https://www.*****.com/us/en/support-downloads/printers.html (zebra)
> 
> adding a TCP/IP print port in Windows (pointing to your sw ip:port ) then using one of the Zebra Print drivers for Win to print?

can you paste the correct link? this one is broken.

**Reply by GozerBCN** [reply]
Somehow the URL got overwritten.
it's at zebra dot com /us/en/etc etc

**Reply by Austinite800** [reply]
This totally works as well (I just found out). If you install a label printer driver like "ZDesigner ZT420 203dpi ZPL" or similar via [Zebra Setup Utilities](https://www.zebra.com/us/en/support-downloads/software/printer-software/printer-setup-utilities.html), you can add a TCP/IP port like 127.0.0.1:9100 during setup and map it to the label printer driver. This setup tool also installs it as a printer in Windows that you can print to while Virtual ZPL Printer is running. 

The method I used below in the word doc is just the way to do it in pure Windows without Zebra Printer Setup Utilities. But, chances are if you are using a virtual ZPL printer, you probably have a real Zebra printer that you'll need to print to anyway, and you'll need the driver for that anyways.

**Reply by rafamuz** [reply]
> This totally works as well (I just found out). If you install a label printer driver like "ZDesigner ZT420 203dpi ZPL" or similar via [Zebra Setup Utilities](https://www.zebra.com/us/en/support-downloads/software/printer-software/printer-setup-utilities.html), you can add a TCP/IP port like 127.0.0.1:9100 during setup and map it to the label printer driver. This setup tool also installs it as a printer in Windows that you can print to while Virtual ZPL Printer is running.
> 
> The method I used below in the word doc is just the way to do it in pure Windows without Zebra Printer Setup Utilities. But, chances are if you are using a virtual ZPL printer, you probably have a real Zebra printer that you'll need to print to anyway, and you'll need the driver for that anyways.

Yeah. the method I used is pure windows. because I don't want to and don't have the zebra utility!

**Comment by Austinite800** [comment]
Hi all, I had the same question. I was able to get this working. Hope you don't mind me sharing what I did!
[Virtual ZPL Printer Setup for Windows.docx](https://github.com/porrey/Virtual-ZPL-Printer/files/14888513/Virtual.ZPL.Printer.Setup.for.Windows.docx)

**Reply by alexwuha** [reply]
Thank you so much!

**Reply by rafamuz** [reply]
I'll try this solution and feedback here. thank you

**Reply by rafamuz** [reply]
Nice. It worked fine.

**Reply by rafamuz** [reply]
> Hi all, I had the same question. I was able to get this working. Hope you don't mind me sharing what I did! [Virtual ZPL Printer Setup for Windows.docx](https://github.com/porrey/Virtual-ZPL-Printer/files/14888513/Virtual.ZPL.Printer.Setup.for.Windows.docx)

Hope the author of the post mark it as answer, cause it's the answer

**Reply by devarajhansa** [reply]
Worked nicely. Thank you
